### PR TITLE
Add the abily to compare Emoji and PartialEmoji classes through string representation

### PR DESCRIPTION
**Pull request by OrangSquid**
Now two Emoji and PartialEmoji instances are compared by their string representation giving a result other than `False` solving #1413 
This was done by overloading in each class the == operator and != operator and comparing each instance's string representation.

### CODE REVIEW

**Comment by slice** [comment]
In my opinion, I don't think using `str()` is a good way to handle comparisons. Seems hackish to me.

File: discord/emoji.py [hunk 1]
```diff
@@ -75,6 +77,16 @@ def __str__(self):
...
         return '<:%s:%s>' % (self.name, self.id)
 
+    def __eq__(self, other_emoji):
+        if type(other_emoji) == Emoji:
+            return self.__str__() == other_emoji.__str__()
+        return super().__eq__(other_emoji)
+
+    def __ne__(self, other_emoji):
+        if type(other_emoji) == Emoji:
+            return self.__str__() != other_emoji.__str__()
```
[the remark this answers is not in the document]
Comment (OrangSquid) [reply discord/emoji.py]: Thanks for remembering me that I could just call str through that way. I have commited the changes.

**Comment by OrangSquid** [comment]
> In my opinion, I don't think using `str()` is a good way to handle comparisons. Seems hackish to me.
I thought this was the best way to compare Emoji and PartialEmoji without messing arround too much with their structure and possibly breaking other stuff. If you have other ideas on how to do it let me know.

**Comment by bmintz** [comment]
Thinking about the information that the string form of a custom emoji represents, it's not hard to see that `(self.name, self.id, self.animated)` could be used instead.

**Pull request closed without merging**